scripts/lib/CIME/case/case_setup.py

```diff
@@ -50,24 +52,60 @@ def _build_usernl_files(case, model, comp):
             ninst = case.get_value("NINST")
         elif ninst == 1:
             ninst = case.get_value("NINST_{}".format(model))
-        nlfile = "user_nl_{}".format(comp)
-        model_nl = os.path.join(model_dir, nlfile)
-        if ninst > 1:
-            for inst_counter in range(1, ninst+1):
-                inst_nlfile = "{}_{:04d}".format(nlfile, inst_counter)
-                if not os.path.exists(inst_nlfile):
-                    # If there is a user_nl_foo in the case directory, copy it
-                    # to user_nl_foo_INST; otherwise, copy the original
-                    # user_nl_foo from model_dir
-                    if os.path.exists(nlfile):
-                        safe_copy(nlfile, inst_nlfile)
-                    elif os.path.exists(model_nl):
-                        safe_copy(model_nl, inst_nlfile)
-        else:
-            # ninst = 1
-            if not os.path.exists(nlfile):
-                if os.path.exists(model_nl):
-                    safe_copy(model_nl, nlfile)
+        default_nlfile = "user_nl_{}".format(comp)
+        model_nl = os.path.join(model_dir, default_nlfile)
+        user_nl_list = _get_user_nl_list(case, default_nlfile, model_dir)
+        # Note that, even if there are multiple elements of user_nl_list (i.e., we are
+        # creating multiple user_nl files for this component with different names), all of
+        # them will start out as copies of the single user_nl_comp file in the model's
+        # source tree.
+        for nlfile in user_nl_list:
+            if ninst > 1:
+                for inst_counter in range(1, ninst+1):
```

Comment:
Have you run any multiinstance tests to confirm that that functionality is still okay?  Multiinstance with multiple ice sheets?

Reply:
I didn't run tests all the way through the build & run, but I did run through case.setup with multi-instance and verified that the appropriate user_nl files were staged in all cases. I did the following tests both with and without the new `get_user_nl_list` function defined:
- Initial case.setup (so no user_nl files present yet in the case)
- Destination file (e.g., user_nl_cism_gris_0001) already present with some contents: verified that this is not overwritten
- Destination files (e.g., user_nl_cism_gris_0001) absent, but single-instance version present with some contents (e.g., user_nl_cism_gris): verified that the contents from the single-instance version were copied into all multi-instance versions

I checked single instance with multiple ice sheets. I didn't check multi-instance with multiple ice sheets, but note that even with a single ice sheet we now have multiple user_nl_cism files, so my tests of multi-instance did cover the situation where a component has multiple user_nl files.

Reply:
sounds good thanks